### PR TITLE
Add dev version of asset compilation

### PR DESCRIPTION
**Pull request by potiuk**
When we moved www asset compilation from container to the host in #25169,
we removed yarn from the image and the old instructions are no longer
valid. Instead of in-container yarn dev we expect now
to have yarn dev run in the host. It can be done either with
pre-commit (if you have no node/yarn installed in your host, pre-commit
will automatically install both node and yarn in the right versions,
or if you have yarn installed you can run it manually.

The 'start-airflow' instructions now remove the in-container
instructions and explain those two options you have.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: dev/breeze/src/airflow_breeze/commands/developer_commands.py [hunk 1]
```diff
@@ -558,13 +566,20 @@ def static_checks(
         allow_extra_args=True,
     ),
 )
+@click.option(
+    "--dev",
+    help="Run development version of assets compilation - it will not quit and automatically "
+    "recompile assets on-the-flight when they are changed.",
```
Comment (bbovenzi) [inline dev/breeze/src/airflow_breeze/commands/developer_commands.py]:
```suggestion
    "recompile assets on-the-fly when they are changed.",
```

**Comment by pierrejeambrun** [comment]
Great

**Review by pierrejeambrun — APPROVED** [review]
Nice 👍

**Comment by potiuk** [comment]
BTW.  @bbovenzi  @pierrejeambrun  - one comment, the nice thing about using pre-commit is that you do not have to care about maintaining your 'node + yarn' in the right version, so it will be always "working the same" for everyone using `breeze` and `pre-commit` under the hood. 

But the problem with pre-commit is that it captures the outpt and only shows it (even in verbose mode) after it completes, but in this case - it will never complete :) - so you do not see the output of the compiled assets as they change (which I find is a good "sanity check" to see that things are atually being refreshed. You will see it in "webserver -d` output, so it's not a big problem, and you can likely live with it, but I decided to mention also the option of runing `yarn dev` manually as it gives instant feedback about stuff being recompiled.

**Comment by bbovenzi** [comment]
Screen Shot 2022-07-25 at 7 58 12 PM

I wonder if we should add something to the message here so that developers know that its actively running and not stuck?

**Comment by potiuk** [comment]
Yeah, why not.

**Comment by potiuk** [comment]
How about this :stuck_out_tongue:  @bbovenzi  ?

image

**Review by bbovenzi — APPROVED** [review]
lgtm!

**Pull request merged**